server/src/main/java/org/opensearch/extensions/action/ExtensionActionResponse.java

```diff
@@ -52,8 +68,27 @@ public void setResponseBytes(byte[] responseBytes) {
         this.responseBytes = responseBytes;
     }
 
+    /**
+     * Gets the Response bytes as a UTF-8 string
+     *
+     * @return A string representation of the response bytes
+     */
+    public String getResponseBytesAsString() {
```

Comment:
Perhaps we could also provide a method to wrap the `responseBytes `with a `StreamInput` to allow reading in objects using a stream input constructor.

Reply:
Great suggestion! Will do.